autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -1853,16 +1868,17 @@ mod tests {
             panic!("Expected row");
         };
 
-        let expiry2 = row.take_required_cell("connected_at").unwrap().timestamp;
-        assert!(expiry2 > expiry);
+        let ca_expiry2 = row.take_required_cell("connected_at").unwrap().timestamp;
+
+        assert!(ca_expiry2 >= ca_expiry);
 
         for mut cells in row.cells.into_values() {
             let Some(cell) = cells.pop() else {
                 continue;
             };
-            assert_eq!(
-                cell.timestamp, expiry2,
-                "{} cell timestamp should match connected_at's",
+            assert!(
+                cell.timestamp >= ca_expiry2,
+                "{} cell timestamp expiry should exceed connected_at's",
```

Comment:
Is this change necessary? IMHO I prefer the stricter check (ensuring everything was touched).

Reply:
Since the timestamp here is the time that bigtable will be (potentially) cleaning up the cell, we want to make sure that doesn't happen until after the expiration timestamp for the data. So, yes, the time we clean up the data should happen either at or later than the time that we want the data to live.